tool/gc_decode.cpp

```diff
@@ -0,0 +1,62 @@
+#include "IRsend.h"
+#include "IRsend_test.h"
+#include <iostream>
+#include <stdio.h>
+#include <string.h>
+#include <errno.h>
+#include <inttypes.h> /* strtoumax */
+#include <stdbool.h>
+
+
+using namespace std;
+
+
+void str_to_uint16(char *str, uint16_t *res)
+{
+  char *end;
+  errno = 0;
+  intmax_t val = strtoimax(str, &end, 10);
+  if (errno == ERANGE || val < 0 || val > UINT16_MAX || end == str || *end != '\0')
+    return;
+  *res = (uint16_t) val;
+}
+
+
+int main (int argc, char * argv[]) {
+
+
+    if (argc != 2) { 
+        cout << "Use gc_decode [global_code]" << endl;
+	return 0;
+    }
+
+    uint16_t gc_test[250];
```

Comment:
Nitpick/suggestion: Probably best to put the 250 value into a #define etc, and make it larger. e.g. 512 or something. Some GC codes can be very long.